src/time-wait-sync/time-wait-sync.c

```diff
@@ -50,7 +50,7 @@ static void clock_state_release(ClockState *sp) {
 static int clock_state_update(ClockState *sp, sd_event *event);
 
 static int update_notify_run_systemd_timesync(ClockState *sp) {
-        sp->run_systemd_timesync_wd = inotify_add_watch(sp->inotify_fd, "/run/systemd/timesync", IN_CREATE|IN_DELETE_SELF);
+        sp->run_systemd_timesync_wd = inotify_add_watch_and_warn(sp->inotify_fd, "/run/systemd/timesync", IN_CREATE|IN_DELETE_SELF);
         return sp->run_systemd_timesync_wd;
```

Comment:
This function is called from two places and both ignore the return value... I think it should be changed to return `void` and `return` dropped.

The problem with this patch is that (despite the name) `inotify_add_watch_and_warn` logs at error level. But here this is not a fatal error, so we would like to have at most `LOG_WARN`. There are other uses of `inotify_add_watch_and_warn` like `time-wait-sync` where the error is a hard error and should be logged as such. So maybe make `inotify_add_watch_and_warn` take the log level as a parameter?